### PR TITLE
Forms Headless API

### PR DESCRIPTION
**Pull request by victorlaerte**

### CODE REVIEW

**Comment by victorlaerte** [comment]
Hey, @nhpatt can you please take a look before resending it to Brian?
I've tested with DDMFormScreenlet and it seems to render all the fields without any problem.

File: modules/apps/headless/headless-form/headless-form-impl/src/main/java/com/liferay/headless/form/internal/dto/v1_0/util/StructureUtil.java [hunk 1]
```diff
@@ -121,6 +121,38 @@ public static FormStructure toFormStructure(
 		);
 	}
 
+	private static Grid _getGrid(DDMFormField ddmFormField, Locale locale) {
+		String type = ddmFormField.getType();
+
+		if (type.equals("grid")) {
```
Comment (nhpatt) [inline modules/apps/headless/headless-form/headless-form-impl/src/main/java/com/liferay/headless/form/internal/dto/v1_0/util/StructureUtil.java]: invert this check to if !equals return early

File: modules/apps/headless/headless-form/headless-form-impl/src/main/java/com/liferay/headless/form/internal/dto/v1_0/util/StructureUtil.java [hunk 2]
```diff
@@ -121,6 +121,38 @@ public static FormStructure toFormStructure(
...
+			return new Grid() {
+				{
+					columns = transform(
+						_toLocalizedValueMapEntry(ddmFormField, "columns"),
+						entry -> new Column() {
```
Comment (nhpatt) [inline modules/apps/headless/headless-form/headless-form-impl/src/main/java/com/liferay/headless/form/internal/dto/v1_0/util/StructureUtil.java]: extract this to a method _toColumn and again with _toRow

File: modules/apps/headless/headless-form/headless-form-impl/src/main/java/com/liferay/headless/form/internal/dto/v1_0/util/StructureUtil.java [hunk 3]
```diff
@@ -181,6 +174,23 @@ private static Grid _getGrid(DDMFormField ddmFormField, Locale locale) {
 		);
 	}
 
+	private static SuccessPage _getSuccessPage(
+		DDMFormSuccessPageSettings ddmFormSuccessPageSettings, Locale locale) {
+
+		if (ddmFormSuccessPageSettings.isEnabled()) {
```
Comment (nhpatt) [inline modules/apps/headless/headless-form/headless-form-impl/src/main/java/com/liferay/headless/form/internal/dto/v1_0/util/StructureUtil.java]: invert check

File: modules/apps/headless/headless-form/headless-form-impl/src/main/java/com/liferay/headless/form/internal/dto/v1_0/util/StructureUtil.java [hunk 4]
```diff
@@ -298,21 +298,6 @@ private static FormPage _toFormPage(
 		};
 	}
 
-	private static String _toInputControl(DDMFormField ddmFormField) {
```
Comment (nhpatt) [inline modules/apps/headless/headless-form/headless-form-impl/src/main/java/com/liferay/headless/form/internal/dto/v1_0/util/StructureUtil.java]: 👍

**Pull request closed without merging**